Service/Shipment/Packingslip/Compatibility/OrderShipmentFactoryProxy.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ *
+ *          ..::..
+ *     ..::::::::::::..
+ *   ::'''''':''::'''''::
+ *   ::..  ..:  :  ....::
+ *   ::::  :::  :  :   ::
+ *   ::::  :::  :  ''' ::
+ *   ::::..:::..::.....::
+ *     ''::::::::::::''
+ *          ''::''
+ *
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Creative Commons License.
+ * It is available through the world-wide-web at this URL:
+ * http://creativecommons.org/licenses/by-nc-nd/3.0/nl/deed.en_US
+ * If you are unable to obtain it through the world-wide-web, please send an email
+ * to servicedesk@tig.nl so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade this module to newer
+ * versions in the future. If you wish to customize this module for your
+ * needs please contact servicedesk@tig.nl for more information.
+ *
+ * @copyright   Copyright (c) Total Internet Group B.V. https://tig.nl/copyright
+ * @license     http://creativecommons.org/licenses/by-nc-nd/3.0/nl/deed.en_US
+ */
+namespace TIG\PostNL\Service\Shipment\Packingslip\Compatibility;
+
+use Magento\Framework\ObjectManagerInterface;
+
+// @codingStandardsIgnoreFile
+class OrderShipmentFactoryProxy
```

Comment:
I mean to use Magento 2 Proxies rather than Proxy class.

This is what I am talking about - https://devdocs.magento.com/guides/v2.3/extension-dev-guide/proxies.html

Reply:
I did already understand that. But one could hard-code a Proxy class yourself as well, just like you hard-code a Factory: It is just a design pattern to inject the ObjectManager, duplicate all target methods and postpone initializing the target until the method is called. (We are not talking about the Proxy design pattern of the GoF, but the one that Magento invented.)